### PR TITLE
Staking: Init pool, add stake

### PR DESCRIPTION
**Pull request by Tristyn**
New style for the lib. Really happy how it's turning out. Especially stakeAccount.ts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@jet-lab/jet-engine",
-  "version": "0.2.2",
+  "version": "0.2.3",
```
Comment (callensm) [inline package.json:3]:
```suggestion
  "version": "0.2.2",
```
This should be changed by `npm version patch` which also pushes a new tag to the repository once PR has landed on `master`.

**Review by callensm — CHANGES_REQUESTED** [review]

File: src/common/index.ts [hunk 2]
```diff
@@ -1,2 +1,35 @@
+import { PublicKey } from "@solana/web3.js"
+import { DerivedAccount } from ".."
```
Comment (qiqicodes) [inline src/common/index.ts:2]: DerivedAccount doesn't exist currently.

**Review by qiqicodes — APPROVED** [review]
LGTM =]

**Comment by Tristyn** [comment]
This was my first rebase :)

**Comment by Tristyn** [comment]
For some reason I accepted Matts change but it's the change is still not accepted

**Review by callensm — APPROVED** [review]

**Pull request merged**